storage/remote/otlptranslator/prometheus/normalize_name.go

```diff
@@ -120,18 +119,19 @@ func BuildCompliantName(metric pmetric.Metric, namespace string, addMetricSuffix
 	return metricName
 }
 
+var nonMetricNameCharRE = regexp.MustCompile(`[^a-zA-Z0-9:]`)
```

Comment:
Noting in passing: this (pulling the initialization out to start of program) should also be done for the other two `regexp.MustCompile` in this file.  Check that relevant benchmarks exist, since they ought to be showing a performance hit.

Reply:
Thanks for the tip @bboreham.

Reply:
I've made the changes, and am benchmarking to see if an effect can be seen.

Reply:
Opened a PR with your suggested optimization @bboreham - simple, but effective.